BibBuddy/app/src/main/java/de/bibbuddy/ImportBibTex.java

```diff
@@ -0,0 +1,339 @@
+
+        if (currentBibItem.startsWith(currentBibTag)) {
+          parsedBibValue = replaceLast(replaceLast(currentBibItem,
+              BibTexKeys.COMMA_SEPARATOR), "\n");
+          parsedBibValue = parsedBibValue.replace(currentBibTag, "");
+          bibItemsList.set(i, parsedBibValue);
+        }
+
+      }
+
+      bibTagValue.put(currentBibTag, parsedBibValue);
+    }
+
+  }
+
+  /** Checks for the position of the next BibTeX tag.
+   * The BibTeX tags (without "@book") do not have fixed order, so this
+   * method stores the positions of the BibTeX tags  in a List and then
+   * through sorting of this list - determines which is the next BibTeX tag.
+   *
+   * @param bibItem  a single BibTeX item as String
+   * @return         the next BibTeX element with a BibTeX tag
+   */
+  public List<String> checkNextBibTag(@NonNull String bibItem) {
+    ArrayList<Integer> nextBibTagIndex = new ArrayList<>();
+    ArrayList<String> bibItemsWithTags = new ArrayList<>();
+
+    if (bibItem.contains(BibTexKeys.BOOK_TAG)) {
+      nextBibTagIndex.add(bibItem.indexOf(BibTexKeys.BOOK_TAG));
+
+      for (String bibTag : bibTags) {
+        if (bibItem.contains(bibTag)) {
+          nextBibTagIndex.add(bibItem.indexOf(bibTag));
+        }
+      }
+
+    }
+
+    nextBibTagIndex.add(bibItem.length());
+    Collections.sort(nextBibTagIndex);
+
+    int currentBibTagPosition = 0;
+    for (int i = 0; i < nextBibTagIndex.size(); i++) {
+      int nextBibTagPosition = nextBibTagIndex.get(i);
+      if (nextBibTagPosition > 0) {
+        bibItemsWithTags.add(bibItem.substring(currentBibTagPosition,
+            nextBibTagPosition));
+        currentBibTagPosition = nextBibTagPosition;
+      }
+    }
+
+    return bibItemsWithTags;
+  }
+
+  /** Parses the authors from the BibTeX content.
+   * Since the author names in BibTeX are comma separated
+   * (no matter how many authors there are), this method
+   * considers parsing of one author and of multiple authors.
+   *
+   * @return         List of parsed author(s) name(s) and family
+   *                 name(s).
+   */
+
+  public List<Author> parseAuthorNames() {
+
+    String authorNames = bibTagValue.get(BibTexKeys.AUTHOR);
+    List<Author> authors = new ArrayList<>();
+
+    // if multiple authors
+    if (authorNames.contains(BibTexKeys.AND_MULTIPLE_AUTHORS)) {
+      String[] names = authorNames.split(BibTexKeys.AND_MULTIPLE_AUTHORS);
+
+      for (String name : names) {
+        getAuthorNames(name, authors);
+      }
+
+    }
+
+    // if one author
+    if (!authorNames.contains(BibTexKeys.AND_MULTIPLE_AUTHORS)) {
+      getAuthorNames(authorNames, authors);
+    }
+
+    return authors;
+  }
+
+  private void getAuthorNames(String authorNames, List<Author> authors) {
+
+    // if the names are comma separated
+    if (authorNames.contains(", ")) {
+      String[] authorName = authorNames.split(", ");
+      authors.add(new Author(authorName[0], authorName[1], ""));
```

Comment:
I'm so sorry Silvia in my own code suggestion with the author fix I made a mistake. Your version was correct. I probably tested it with an old bibtex file so the sequence of the authors was messed up.
```suggestion
      authors.add(new Author(authorName[1], authorName[0], ""));
```